.github/workflows/rustdoc.yml

```diff
@@ -34,6 +34,16 @@ jobs:
         with:
           repo-token: ${{ secrets.GITHUB_TOKEN }}
 
+      - name: Configure cache
+        uses: actions/cache@704facf57e6136b1bc63b828d79edcd491f0ee84 # @v3.3.2
+        with:
+          path: |
+            ~/.cargo/registry
+            ~/.cargo/git
+          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.toml') }}
+          restore-keys: |
+            ${{ runner.os }}-cargo-
```

Comment:
Why not search for the key first then try to pattern match after?

i.e:
```
 restore-keys: |
            ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.toml') }}
            ${{ runner.os }}-cargo-
            ${{ runner.os }}
```

Reply:
`restore-keys` is for additional keys, `key` is still considered first. I had it too initially, but it is redundant, so I removed it.